tests/integration/client/feedback/dataset/remote/test_dataset.py

```diff
@@ -841,3 +842,51 @@ async def test_warning_local_methods(self, role: UserRole) -> None:
                 match="A local `FeedbackDataset` returned because `prepare_for_training` is not supported for `RemoteFeedbackDataset`. ",
             ):
                 ds.prepare_for_training(framework=None, task=None)
+
+    async def test_add_records_with_metadata_including_nan_values(
+        self, owner: "User", feedback_dataset: FeedbackDataset
+    ):
+        argilla.client.singleton.init(api_key=owner.api_key)
+        workspace = Workspace.create(name="test-workspace")
+
+        remote_dataset = feedback_dataset.push_to_argilla(name="test_dataset", workspace=workspace)
+
+        records = [
+            FeedbackRecord(
+                external_id=str(i),
+                fields={"text": "Hello world!", "text-2": "Hello world!"},
+                metadata={"float-metadata": float("nan")},
+            )
+            for i in range(1, 20)
+        ]
+        remote_dataset.add_records(records)
+
+        assert len(remote_dataset.records) == len(records)
+        assert all("float-metadata" not in record.metadata for record in remote_dataset.records)
+
+    async def test_add_records_with_metadata_including_none_values(
```

Comment:
How would the `None` `np.nan` difference work within the UI. Also, what happens if we upload `np.nan` to a `TermsMetadataProperty` or `IntMetadataProperty`.

Reply:
`np.nan` will be ignored. `None` values will be accepted as `None`. I would not assume a similar treatment between `nan` and `None` values. This is the reason of the difference